### ISSUE TITLE
on:click:value:to not working anymore for buttons since CanJS 4

### ISSUE THREAD

**Issue by Mamuuut** [canjs/can-stache-bindings#484]
The following stache template is not working the same way as in CanJS v3:
`<button value="1" on:click:value:to="myScopeProp"></button>`

I think there are 2 issues:
1. the value is not supported anymore for button because value is has special attr rule test only matching form element:
`var formElements = {"INPUT": true, "TEXTAREA": true, "SELECT": true}`
Is this new behavior expected in CanJS v5?
2. If I replace value by a different attribute name, the scope property is updated when the binding is created and not only when the click event is triggered.

Here is the working version in CanJS v3:
https://codepen.io/Mamuuut/pen/ZMaOzG

And the broken one in v4:
https://codepen.io/Mamuuut/pen/NLwrrV

**Comment by justinbmeyer** [comment]
Is value a supported/native property of a button?

**Comment by Mamuuut** [comment]
It seems to be:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button
https://html.spec.whatwg.org/multipage/form-elements.html#the-button-element

**Comment by justinbmeyer** [comment]
That formElements object is wrong then.

**Comment by Mattchewone** [comment]
Discussion regarding fixing the issue where initialisation sets the `value` and then when the **click** event happens the value hasn't changed since initialisation so nothing happens,  between @justinbmeyer @phillipskevin @Mattchewone :
```
<button value="5" on:input:value:to="foo" on:rendered:value:to="foo"/>
<button value="4" on:input:value:to="foo"/>

var leftEl = new AttributeObservable(el,"value","click");


var rightFoo = new ScopeKeyData(scope,"foo");


// new Bind({child: leftEl, parent: rightFoo, updateParentWhen: onClicks})

leftEl.on(function(newVal){
    rightFoo.set(newVal);
})

1.  Initialize `leftEl` to empty -> pushes the problem down the road X

2.  Produce an event each time ... CanJS expects this ... what have done ...?

3.  Stream VS Property construct in CanJS ....
        onEvent( .... )?

        - AttributeObservable ... delete onValue / offValue ... only implements `onEvent`
        - can-bind ... if given something only with `onEvent`, listen to `onEvent` to update sibling
        -

4.  {getValue: function(){ return "5"}} -> how many things would have to deal with this

5.  When to update observable ...
        new Bind({child: leftEl, parent: rightFoo, updateParentWhen: onClicks})

        - makes can-stache-bindings ugly
```

**Comment by Mattchewone** [comment]
Based on having implemented **`#2`** `Produce an event each time` and have this passing all tests.

Now looking into the `AttributeObservable` using `onEmit` / `offEmit` when we have an `event` rather than using `onValue` / `offValue`. 

For the test for `AttributeObservable` it would be something like this  - 

```js
testIfRealDocument("can use onEmit symbol with event", function(assert) {
  var done = assert.async(2);
  var button = document.createElement("button");
  button.value = 5;

  var ta = this.fixture;
  ta.appendChild(button);

  var obs = new AttributeObservable(button, "value", {}, "click");
  
  // Listen to emitted values
  obs[canSymbol.for('can.onEmit')](function (val) {
    assert.equal(val, 5);
    done();
  });

  // Dispatch two click's
  domEvents.dispatch(button, "click");
  domEvents.dispatch(button, "click");
});
```

with `AttributeObservable` using it like :
```js
this[canSymbol.for('can.onEmit')] = function(callback, queue) {
	AttributeObservable.prototype.on.call(this, callback, queue);
}
```

Then within `can-bind` we would need to check for this `can.onEmit` Symbol using a similar `updateValue` process but we would want to compare the existing `parent`/`child` with the `newVal` before setting it's value.
